src/graph_compute/concurrent_queue.h

```diff
@@ -0,0 +1,48 @@
+#ifndef CONCURRENT_QUEUE_H_
+#define CONCURRENT_QUEUE_H_
+
+#include "../libgraphio/libgraphio.h"
+
+struct mrmw_queue_t {
```

Comment:
It would be nice to walk through this code in person in the next meeting, it's very complex to read in isolation.